block/manager.go

```diff
@@ -104,9 +105,33 @@ type Manager struct {
 
 // getInitialState tries to load lastState from Store, and if it's not available it reads GenesisDoc.
 func getInitialState(store store.Store, genesis *cmtypes.GenesisDoc) (types.State, error) {
+	// Load the state from store.
 	s, err := store.GetState(context.Background())
-	if err != nil {
+	if err == ds.ErrNotFound {
```

Comment:
this should be `errors.Is(err, ds.ErrNotFound)` so that you can take advantage for the golang errors package wrapping. 

https://pkg.go.dev/errors